docs/adr/0009-naming-conventions.md

```diff
@@ -0,0 +1,126 @@
+# Naming conventions
+
+Date: 2020-09-24
+
+## Status
+
+2020-12-02 Proposed
+
+## Component files
+
+### Principles
+
+- Keep it simple, reduce nesting
+- 'container'-type components in their own folder directly under `src`.
+- Components needed only by one 'root' component go into that root component's folder, no extra nesting
+- Components needed by more than one 'root' component go into `src/components`
+- Component folders, files, and auxiliary files, camel cased, starting with upper case
+- Component names and file names will be unique
+- Unique `data-testid` names by prefixing them with component names
+- 'root' level components, hooks, interfaces, services, in their respective floders.
+
+### Example tree
+
+```
+src
+  App
+    App.tsx
+    App.styles.ts
+    App.test.ts
+  DefaultTexts
```

Comment:
I feel like having all container-type components at the root level will pollute the `src` directory. Why not keep a `containers` directory at this level?

Reply:
Yes, that might be a good idea.

Reply:
I don't like marking them as `containers` though. It's the structure of the application.
How about `src/app`? Or even `src/App` and put the `App` component files directly in there.